capreolus/extractor/bertpassage.py

```diff
@@ -223,7 +223,35 @@ def parse_label_tensor(x):
 
         return (pos_bert_input, pos_mask, pos_seg, neg_bert_input, neg_mask, neg_seg), label
 
-    def _prepare_doc_psgs(self, doc):
+    def _get_passages(self, docid):
```

Comment:
So we look up the docid in the anserini index on every iteration right? This should be fine. I think the reason why we stored the entire `docid2passages` in memory was so that during training PyTorch won't lose time tokenizing the text. But since we are anyway pre-computing TFRecords, looking these up from the index makes sense :+1: